tests/shard_map_test.py

```diff
@@ -2162,7 +2162,22 @@ def f():
                        mesh, in_specs=P('i', None), out_specs=P('i', None),
                        check_rep=False, auto=frozenset({'j'}))()
 
-    self.assertAllClose(f(), np.array(range(4), dtype=np.int32).reshape(-1, 1))
+    self.assertAllClose(f(), np.arange(4, dtype=np.int32).reshape(-1, 1))
+
+  def test_partial_auto_axis_index_degenerated_axis(self):
+    if config.use_shardy_partitioner.value:
+      self.skipTest('Shardy does not support full-to-shard.')
+
+    mesh = jtu.create_mesh((1, 8), ('i', 'j'))
```

Comment:
Do you mind using `(1, 2)` mesh so that we get more test coverage?

Reply:
Done.

This branch is rebased and squashed.